### ISSUE TITLE
git merge-base fails in Jenkins Multi-branch workspace when branch becomes not ff-mergable

### ISSUE THREAD

**Issue by astj** [haya14busa/reviewdog#103]
[reviewdog get diffs using `git merge-base`](https://github.com/haya14busa/reviewdog/blob/master/github.go#L179-L192), but in Jenkins multi-branch workspace, this fails in some condition (and I think reviewdog can do nothing about it...).

# Situation

Jenkins multi-branch plugin fetches only target branch by following `git fetch`, which only obtains target branch's change.
```
git fetch --no-tags --progress git@myserver:myrepository.git +refs/heads/MyBranch:refs/remotes/origin/MyBranch
```
And reviewdog determines base commit using [github pull request api](https://github.com/haya14busa/reviewdog/blob/master/github.go#L172-L178).

In following situation the base commit obtained by github api does not exist in workspace, therefore `git merge-base` fails.
- MyBranch, which is ff-mergable to BaseBranch is pushed to Jenkins
  - at this time, BaseBranch's HEAD is exists in workspace so reviewdog can work properly
- Some new commit is added to BaseBranch
- MyBranch is pushed again
  - at this time, MyBranch is not ff-mergable to BaseBranch
  - jenkins only obtains changes in MyBranch by `git fetch`, so new commits in BaseBranch does not exist in this workspace 
  - reviewdog tries to `git merge-base`, but fails because HEAD of BaseBranch does not exist in the workspace.

# Possible solution

I think reviewdog can do nothing about it 😭 
Jenkins, or users should call `git fetch BaseBranch` before calling reviewdog.

**Comment by haya14busa** [comment]
Thank you for the report!

Hmm... maybe I should revisit #66.
If the diff provided by GitHub can be used, we don't need to call "git diff" locally in CI  in the first place.

**Comment by haya14busa** [comment]
Fixed by #105  :sparkles:
Please update reviewdog version >= 0.9.8
